python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py

```diff
@@ -517,6 +522,99 @@ def load(self, checkpoint, **kwargs):
+             options=None):
+        """
+        Saves the model to Tensorflow SavedModel or a single HDF5 file.
+
+        :param filepath: String, PathLike, path to SavedModel or H5 file to save the
+            model. It can be local/hdfs/s3 filepath
+        :param overwrite: Whether to silently overwrite any existing file at the
+            target location, or provide the user with a manual prompt.
+        :param include_optimizer: If True, save optimizer's state together.
+        :param save_format: Either `'tf'` or `'h5'`, indicating whether to save the
+            model to Tensorflow SavedModel or HDF5. Defaults to 'tf' in TF 2.X,
+            and 'h5' in TF 1.X.
+        :param signatures: Signatures to save with the SavedModel. Applicable to the
+            'tf' format only. Please see the `signatures` argument in
+            `tf.saved_model.save` for details.
+        :param options: (only applies to SavedModel format)
+            `tf.saved_model.SaveOptions` object that specifies options for
+            saving to SavedModel.
+        """
+        # get current trained model
+        model = self.get_model()
+
+        if is_local_path(filepath):
+            model.save(filepath, overwrite, include_optimizer, save_format, signatures, options)
+        else:
+            file_name = os.path.basename(filepath)
+            temp_dir = tempfile.mkdtemp()
+            temp_path = os.path.join(temp_dir, file_name)
+            try:
+                model.save(temp_path, overwrite, include_optimizer, save_format, signatures,
+                           options)
+                if save_format == 'h5' or filepath.endswith('.h5') or filepath.endswith('.keras'):
+                    # hdf5 format
+                    put_local_file_to_remote(temp_path, filepath, over_write=overwrite)
+                else:
+                    # tf format
+                    put_local_dir_tree_to_remote(temp_path, filepath)
+            finally:
+                shutil.rmtree(temp_dir)
+
+    def load(self,
+             filepath,
+             custom_objects=None,
+             compile=True):
+        """
+        Loads a model saved via `estimator.save()
+
+        :param filepath: (str) Path of saved model.
+        :param custom_objects: Optional dictionary mapping names
+          (strings) to custom classes or functions to be
+          considered during deserialization.
+        :param compile: Boolean, whether to compile the model after loading.
+        :param options: Optional `tf.saved_model.LoadOptions` object that specifies
+        options for loading from SavedModel.
+
+        """
+        if is_local_path(filepath):
+            params = dict(
+                filepath=filepath,
+                custom_objects=custom_objects,
+                compile=compile
+            )
+            model_refs = [
+                worker.load_model.remote(**params) for worker in self.remote_workers
+            ]
+            ray.get(model_refs)
```

Comment:
is this logic correct? Will the filepath be available to all workers?
Should we first load to the driver and send the model to all workers?

Reply:
> is this logic correct? Will the filepath be available to all workers? Should we first load to the driver and send the model to all workers?

Yes, it's available. Load to driver maybe not a good solution, since the model will be created in `TFRunner`.